lnwire/fuzz_test.go

```diff
@@ -0,0 +1,726 @@
+		}
+
+		if first.HtlcMinimum != second.HtlcMinimum {
+			shouldPanic = true
+		}
+
+		if first.FeePerKiloWeight != second.FeePerKiloWeight {
+			shouldPanic = true
+		}
+
+		if first.CsvDelay != second.CsvDelay {
+			shouldPanic = true
+		}
+
+		if first.MaxAcceptedHTLCs != second.MaxAcceptedHTLCs {
+			shouldPanic = true
+		}
+
+		if !first.FundingKey.IsEqual(second.FundingKey) {
+			shouldPanic = true
+		}
+
+		if !first.RevocationPoint.IsEqual(second.RevocationPoint) {
+			shouldPanic = true
+		}
+
+		if !first.PaymentPoint.IsEqual(second.PaymentPoint) {
+			shouldPanic = true
+		}
+
+		if !first.DelayedPaymentPoint.IsEqual(second.DelayedPaymentPoint) {
+			shouldPanic = true
+		}
+
+		if !first.HtlcPoint.IsEqual(second.HtlcPoint) {
+			shouldPanic = true
+		}
+
+		if !first.FirstCommitmentPoint.IsEqual(second.FirstCommitmentPoint) {
+			shouldPanic = true
+		}
+
+		if first.ChannelFlags != second.ChannelFlags {
+			shouldPanic = true
+		}
+
+		if !bytes.Equal(first.UpfrontShutdownScript, second.UpfrontShutdownScript) {
+			shouldPanic = true
+		}
+
+		if shouldPanic {
+			t.Fatal("original message and deserialized message are not equal")
+		}
+
+	})
+}
+
+func Fuzz_ping(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		// Prefix with MsgPing.
+		data = prefixWithMsgType(data, MsgPing)
+
+		// Pass the message into our general fuzz harness for wire messages!
+		harness(t, data)
+	})
+}
+
+func Fuzz_pong(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		// Prefix with MsgPong.
+		data = prefixWithMsgType(data, MsgPong)
+
+		// Pass the message into our general fuzz harness for wire messages!
+		harness(t, data)
+	})
+}
+
+func Fuzz_query_channel_range(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		// Prefix with MsgQueryChannelRange.
+		data = prefixWithMsgType(data, MsgQueryChannelRange)
+
+		// Pass the message into our general fuzz harness for wire messages!
+		harness(t, data)
+	})
+}
+
+func Fuzz_query_short_chan_ids_zlib(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		var buf bytes.Buffer
+		zlibWriter := zlib.NewWriter(&buf)
+		_, err := zlibWriter.Write(data)
+		if err != nil {
+			// Zlib bug?
+			t.Fatal(err)
+		}
+
+		if err := zlibWriter.Close(); err != nil {
+			// Zlib bug?
+			t.Fatal(err)
+		}
+
+		compressedPayload := buf.Bytes()
+
+		chainhash := []byte("00000000000000000000000000000000")
+		numBytesInBody := len(compressedPayload) + 1
+		zlibByte := []byte("\x01")
+
+		bodyBytes := make([]byte, 2)
+		binary.BigEndian.PutUint16(bodyBytes, uint16(numBytesInBody))
+
+		payload := append(chainhash, bodyBytes...)
+		payload = append(payload, zlibByte...)
+		payload = append(payload, compressedPayload...)
+
+		// Prefix with MsgQueryShortChanIDs.
+		payload = prefixWithMsgType(payload, MsgQueryShortChanIDs)
+
+		// Pass the message into our general fuzz harness for wire messages!
+		harness(t, payload)
+	})
+}
+
+func Fuzz_query_short_chan_ids(f *testing.F) {
```

Comment:
im sure the go people have their reasons, but imo the way they did this is dumb. you can't have fuzz tests with similar prefixes, so this one has to have a name change so that running `go test -run=^$ -fuzz=Fuzz_query_short_chan_ids` actually works

Reply:
it seems like this is fine with `go test -run=^$ -fuzz=^Fuzz_reply_channel_range\$`

Reply:
Will still change so no regex needed

Reply:
Thought about putting _no_zlib at end of one
or moving zlib ahead in the prefix of the other
latter seems less awkward